podpac/core/cache/cache.py

```diff
@@ -679,11 +808,283 @@ def has(self, node, key, coordinates=None):
+        
+        Parameters
+        ----------
+        node : podpac.core.node.Node
+        key : str, CacheWildCard
+            CacheWildCard indicates to match any key
+        coordinates : podpac.core.coordinates.coordinates.Coordinates, CacheWildCard, None
+            CacheWildCard indicates to match any coordinates
+        
+        Returns
+        -------
+        TYPE : str
+            Fileglob of existing paths that match the request
+        """
+        pre = '*'
+        nKeY = 'nKeY{}'.format(self.hash_node(node))
+        kKeY = 'kKeY*' if isinstance(key, CacheWildCard) else 'kKeY{}'.format(self.cleanse_filename_str(self.hash_key(key)))
+        cKeY = 'cKeY*' if isinstance(coordinates, CacheWildCard) else 'cKeY{}'.format(self.hash_coordinates(coordinates))
+        filename = '_'.join([pre, nKeY, kKeY, cKeY])
+        filename = filename + '.' + self._extension
+        return glob(self._path_join([self.cache_dir(node), filename]))
+
+    def clear_entire_cache_store(self):
+        shutil.rmtree(self._root_dir_path)
+        return True
+
+    def dir_is_empty(self, directory):
+        return os.path.exists(directory) and os.path.isdir(directory) and not os.listdir(directory)
+
+    def rem_dir(self, directory):
+        os.rmdir(directory)
+
+class S3CacheStore(FileCacheStore):
+
+    _delim = '/'
+
+    def __init__(self, root_cache_dir_path=None, storage_format='pickle', 
+                 s3_bucket=None, aws_region_name=None, aws_access_key_id=None, aws_secret_access_key=None):
+        """Initialize a cache that uses a folder on a local disk file system.
+        
+        Parameters
+        ----------
+        root_cache_dir_path : None, optional
+            Root directory for the files managed by this cache. `None` indicates to use the folder specified in the global podpac settings. Should be the common "root" s3 prefix that you want to have all cached objects stored in. Do not store any objects in the bucket that share this prefix as they may be deleted by the cache.
+        storage_format : str, optional
+            Indicates the file format for storage. Defaults to 'pickle' which is currently the only supported format.
+        s3_bucket : str, optional
+            bucket name, overides settings
+        aws_region_name : str, optional
+            e.g. 'us-west-1', 'us-west-2','us-east-1'
+        aws_access_key_id : str, optional
+            overides podpac settings if both `aws_access_key_id` and `aws_secret_access_key` are specified
+        aws_secret_access_key : str, optional
+            overides podpac settings if both `aws_access_key_id` and `aws_secret_access_key` are specified
+        
+        Raises
+        ------
+        CacheException
+            Description
+        e
+            Description
+        NotImplementedError
+            If unsupported `storage_format` is specified
+        """
+        if boto3 is None:
```

Comment:
This is no longer needed. Lazy_module will handle this for us with a sane error message.

Reply:
thanks.